driver: Enable AIE preemption

NishadSaraf opened the pull request:
Enable AIE preemption and add interface to force preemption.

maxzhen commented:
I don't see you update xrt submodule. Do you have any dependency on newer version of XRT?

NishadSaraf commented:
> I don't see you update xrt submodule. Do you have any dependency on newer version of XRT?

This PR doesn't depend on XRT latest. The existing version works fine.

mamin506 commented on src/driver/amdxdna/npu4_regs.c:
```diff
@@ -12,6 +12,8 @@ const struct rt_config npu4_default_rt_cfg[] = {
 	{ 2, 1, AIE2_RT_CFG_CLK_GATING }, /* Clock gating on */
 	{ 3, 1, AIE2_RT_CFG_CLK_GATING }, /* Clock gating on */
 	{ 4, 1, AIE2_RT_CFG_CLK_GATING }, /* Clock gating on */
+	{ 12, 1, AIE2_RT_CFG_FINE_PREEMPTION },
+	{ 13, 0, AIE2_RT_CFG_FORCE_PREEMPTION },
```
Add comment like other entries.

mamin506 reviewed (changes requested):
The two new RT config is not in NPU1. This is okay and good. aie2_runtime_cfg() will skip it on NPU1.

But the ndev->force_preempt_enabled flag was set for NPU1 as well. And it can be get by ioctl(). If this is not supported in NPU1, please keep ndev->force_preemption_enabled as 0" to avoid confusing.

NishadSaraf commented:
> Quote reply
> Reference

Hi @mamin506, Thanks for the quick review. I have addressed all the review comments. Please let me know if missed anything. Thanks :)

mamin506 reviewed (approved):
LGTM

maxzhen commented:
retest this please

maxzhen commented:
retest this please

mamin506 commented:
retest this please
